daemon/api_snaps.go

```diff
@@ -587,6 +606,11 @@ func snapOpMany(c *Command, r *http.Request, user *auth.UserState) Response {
 	if inst.Channel != "" || !inst.Revision.Unset() || inst.DevMode || inst.JailMode || inst.CohortKey != "" || inst.LeaveCohort || inst.Purge {
 		return BadRequest("unsupported option provided for multi-snap operation")
 	}
+
+	if err := inst.validateSnapshotOptions(); err != nil {
```

Comment:
Thanks for this comment, this matches my initial thinking. However, having `validateSnapshotOptions` not inside `validate` is intentional, would appreciate your view on my reasoning.

Reasoning:
- Validate is shared between POST `/v2/snaps/[name]` POST `/v2/snaps` but
- only `/v2/snaps/[name]` does not support snapshot, and
- only checks allowed commands after running `validation` when it gets to ` dispatch`
- which means this flow will provide feedback on invalid use of snapshot options before saying action snapshot is not allowed

So from my perspective the principle concern is that it opens up the risk of exercising code that is arguably irrelevant to POST `/v2/snaps/[name]`. In the spirit of not fiddling with the existing proven code, I felt lowest risk, simplest option is to only exercise snapshot option validation for `/v2/snaps`.

Your thoughts?

Reply:
The line above this one filters options that aren't supported in the multi-snap operation (e.g., things like `revision` or `channel` that don't make sense for multiple snaps). The handler for the single snap operation can do the same. It makes sense to me that the instruction's `validate()` should validate its fields, independently of its callers and what they might want to do on top of it

Reply:
I though about this some more: My underlying concern is more a matter of priority/order of checking. If I include `validateSnapshotOptions` within `validate`, it will complain about about options of the snapshot action where it would otherwise have said `snapshot action not supported`. And if that check would happen first, the need for further validation falls away.

This is different from the case at line 606 in a subtle way. The action is relevant to both single and multi POSTS, just the options is not.

I do not feel strongly about this one, so just let me know if you still prefer the change?

Reply:
Even in this method, the "unsupported snap operation" check happens after all other validation. So the existing code is already not keeping that order strictly because it would be too complicated to do it. Besides that these errors would be triggered by programmer error/API misusage so in practice they're not going to be seen often. My personal preference would be to to simplify the code

Reply:
Ok, thanks, I move it into validate()